src/wallet/wallet2.cpp

```diff
@@ -559,6 +555,27 @@ void wallet2::check_acc_out_precomp(const crypto::public_key &spend_public_key,
   error = false;
 }
 //----------------------------------------------------------------------------------------------------
+void wallet2::check_acc_out_precomp_onetime(const crypto::public_key &spend_public_key, const tx_out &o, const crypto::key_derivation &derivation, const crypto::secret_key* onetime_h, size_t i, uint64_t &received, uint64_t &money_transfered, bool &error) const
+{
+  received = 0;
+  bool received_bool;
+  check_acc_out_precomp(spend_public_key, o, derivation, i, received_bool, money_transfered, error);
+  if (received_bool)
+  {
+    received = 1;
+  }
+  else if (onetime_h)
+  {
+    crypto::public_key spend_public_key2;
+    crypto::key_derivation derivation2;
+    crypto::secret_key_mult_public_key(*onetime_h, spend_public_key, spend_public_key2);
+    crypto::secret_key_mult_public_key(*onetime_h, reinterpret_cast<const crypto::public_key&>(derivation), reinterpret_cast<crypto::public_key&>(derivation2));
+    check_acc_out_precomp(spend_public_key2, o, derivation2, i, received_bool, money_transfered, error);
+    if (received_bool)
+      received = 2;
```

Comment:
I'd rather see two booleans (received, and onetime, eg), makes things a lot more understandable